.teamcity/_self/PluginBaseBuild.kt

```diff
@@ -81,7 +81,12 @@ open class PluginBaseBuild : Template({
 				# Update composer
 				composer install
 
-				$yarn_install_cmd
+				# Prepare required packages in the monorepo
+				yarn workspaces focus @automattic/calypso-color-schemes
+				yarn workspace @automattic/calypso-color-schemes prepare
```

Comment:
I'm a little unsure on this change. I think it would be hard to run a focused workspace locally -- are there other workspaces which might need to be focused as part of builds in the future? Do we need to document all of them?

Reply:
I see your point. I'd like to keep it as simple as possible and make the changes requires so we just need `yarn workspaces focus` once. But I'm not sure how we can do that.

We have two cases in our repo where we have needed this "double focus": here and in e2e tests:

- Here we need it because of how `@automattic/calypso-color-schemes` works. It has the source files, and scripts to compile those files into some distribution files that are included in the final package. As such, we need to run some compilation process before publishing it (aka `prepare`) _and_ before any other package in the repo can use it.

- E2E tests depend on `@automattica/calypso-e2e`, a TypeScript project. It has to be compiled before it can be imported by Jest tests. Same scenario as before, we need to run something before publishing _and_ before internal usage.

I don't think we should move to a `postinstall` scenario (move the burden of compiling those packages to the consumer, instead of the producer). But other than manually installing and compiling each required workspace, I don't know how we can solve it.

Any ideas?

Reply:
Got an idea in Yarn's discord server: https://discord.com/channels/226791405589233664/654372321225605128/916290658384351312

Reply:
I've implmented that idea @noahtallen ^. Would you mind taking a look?

Reply:
Cool! I think it works. I wish it could auto-detect the package in cwd, though. It would also be nice if this could hook into the focus event.

But I think this is a good starting point because it's generic enough to document for any package, should one want to build anything in an isolated way.